net.go

```diff
@@ -964,3 +988,55 @@ func (m *Memberlist) readUserMsg(bufConn io.Reader, dec *codec.Decoder) error {
 
 	return nil
 }
+
+// sendPingAndWaitForAck makes a TCP connection to the given address, sends
+// a ping, and waits for an ack. All of this is done as a series of blocking
+// operations, given the deadline.
+func (m *Memberlist) sendPingAndWaitForAck(destAddr net.Addr, ping ping, deadline time.Time) (didContact bool, err error) {
```

Comment:
Just a minor style thing, I'd explicitly return a tuple for clarity in each exit path in this function. It's not functionally different than what you have, but it avoids some assignment and to me is a litter easier to read. Up to you.